### DISCUSSION TITLE
part-ii/2021-28/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#361] [Question comments]
Part II 2021 Automata and Formal Languages: Paper 3, Section II, $12 F$ 

https://questions.tripos.org/part-ii/2021-28/

**Comment by artie2000** [comment]
<details>
<summary>Modifying the proof:</summary>

Each nonterminal production has form $A\to B_1\dots B_k$. Let $K$ be the largest such $k$ among nonterminal productions, and set the pumping constant to $K^{m+1}$, where $m$ is the number of nonterminals, instead of the usual $2^{m+1}$.

</details>

<details>
<summary>(i)</summary>

Yes: generated by $S\to aSa\mid\varepsilon\mid B,\ B\to b\mid\varepsilon$.

</details>
<details>
<summary>(ii)</summary>

Yes: generated by $S\to aSb\mid\varepsilon\mid A,\ A\to a\mid\varepsilon$.

</details>
<details>
<summary>(iii)</summary>

No: pump down on $a^nb^nana^nb^n$, where $n$ is the pumping constant.

</details>
